packages/cli/src/InternalHooks.ts

```diff
@@ -266,11 +372,40 @@ export class InternalHooksClass implements IInternalHooksClass {
 			}
 		}
 
-		return Promise.all([
+		promises.push(
+			properties.success
+				? eventBus.sendWorkflowEvent({
+						eventName: 'n8n.workflow.success',
+						payload: {
+							success: properties.success,
+							userId: properties.user_id,
+							workflowId: properties.workflow_id,
+							isManual: properties.is_manual,
+							workflowName: workflow.name,
+						},
+				  })
+				: eventBus.sendWorkflowEvent({
+						eventName: 'n8n.workflow.failed',
+						payload: {
+							success: properties.success,
+							userId: properties.user_id,
+							workflowId: properties.workflow_id,
+							lastNodeExecuted: runData?.data.resultData.lastNodeExecuted,
+							errorNodeType: properties.error_node_type,
+							errorNodeId: properties.error_node_id?.toString(),
+							errorMessage: properties.error_message?.toString(),
+							isManual: properties.is_manual,
+							workflowName: workflow.name,
+						},
+				  }),
+		);
+
+		Promise.all([
 			...promises,
 			BinaryDataManager.getInstance().persistBinaryDataForExecutionId(executionId),
 			this.telemetry.trackWorkflowExecution(properties),
-		]).then(() => {});
+		]).catch((error) => console.log(error));
```

Comment:
we currently block on `persistBinaryDataForExecutionId` and if that throws, the error is propagated up. with these changes, the error now simply gets suppressed.